rocprim/include/rocprim/intrinsics/thread.hpp

```diff
@@ -75,8 +75,7 @@ unsigned int lane_id()
     #ifdef ROCPRIM_HC_API
         return hc::__lane_id();
     #else // HIP
-        // TODO: Find HIP function for that
-        return hc::__lane_id();
+        return __lane_id();
```

Comment:
Does HIP in ROCm 1.8.2 have this function? If not then I am not sure if can merge it to `develop`.

Actually we have branch on our private repo which replaces HC intrinsics with HIP functions (in some places intrinsics just weren't in HIP, in some they didn't work).

Reply:
This function seems to exist on HIP branch roc-1.8.x, I'm not sure if that is specifically ROCm 1.8.2. 

Do you remember which intrinsic were missing or didn't work?

Reply:
* `__shfl*` functions - they **still don't work correctly** on ROCm 1.8.2. From what I see changes you made in https://github.com/ROCm-Developer-Tools/HIP/pull/515 are still only in `master`. I don't know if those functions work correctly in hip-clang. If they don't that may be a reason for many failures.
* `__bitextract_u{32,64}`, `__bitinsert_u32`  - in worst case scenario those can be replaced with some math like it's done for `nvcc`,
* `__lane_id`, `__amdgcn_mbcnt_{lo,hi}` - those are present in ROCm 1.8.2

Reply:
Does rocPRIM have its own rocm 1.8.x branch which can be used on ROCm 1.8.2, and develop used on the master of HIP?

Reply:
No, but we can do it the other way around: have a special branch for changes required for hip-clang and  HIP master.